server.js

```diff
@@ -3,27 +3,85 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
+// MONGOOSE SETUP
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// MODEL FOR THOUGHT
+const Thought = mongoose.model('Thought', {
+  message: { type: String, requierd: true, minlength: 5, maxlength: 140 },
+  hearts: { type: Number, default: 0 },
+  createdAt: { type: Date, default: Date.now }
+})
+
+// SEEDING FOR ADDING NEW DATA
+if (process.env.RESET_DB) {
+  console.log('Resetting database')
+  const seedDatabase = async () => {
+    await Thought.deleteMany({})
+
+    // thought.forEach(() => {
+    //   new Thought().save()
+    // })
+  }
+  seedDatabase()
+}
+
+// PORT & APP SETUP
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailabale' })
+  }
+})
+
+// THOUGHTS
+app.get('/', async (req, res) => {
+  // Returning 20 thoughts order by last created first
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  if (thoughts) {
+    res.status(200).json(thoughts)
+  } else {
+    res.status(404).json({ message: 'Could not find thoughts', error: err.errors })
+  }
+})
+
+// POST A THOUGHT
+app.post('/', async (req, res) => {
```

Comment:
right now it's actually possible to send a happy thought without specifying the message if I'm using Postman. That's because you spelt `required` wrong in your model. Although the frontend is keeping us from posting an "empty" thought you should also check that the backend code is preventing this.